src/org/ods/services/NexusService.groovy

```diff
@@ -88,12 +69,23 @@ class NexusService {
     }
 
     @SuppressWarnings('LineLength')
-    @NonCPS
     URI storeComplextArtifact(String repository, byte[] artifact, String contentType, String repositoryType, Map nexusParams = [ : ]) {
-        def restCall = Unirest.post("${this.baseURL}/service/rest/v1/components?repository={repository}")
-            .routeParam('repository', repository)
-            .basicAuth(this.username, this.password)
+        script.withCredentials([
+            script.usernamePassword(
+                credentialsId: credentialsId,
+                usernameVariable: 'USERNAME',
+                passwordVariable: 'PASSWORD'
+            )
+        ]) {
+            def restCall = Unirest.post("${this.baseURL}/service/rest/v1/components?repository={repository}")
+                .routeParam('repository', repository)
+                .basicAuth(script.env.USERNAME, script.env.PASSWORD)
+            return processsStoreArtifact(restCall, repository, artifact, contentType, repositoryType, nexusParams)
+        }
+    }
 
+    @NonCPS
+    private URI processsStoreArtifact(def restCall, String repository, byte[] artifact, String contentType, String repositoryType, Map nexusParams = [ : ]) {
```

Comment:
There's a typo, one more s than needed. I would also call it processStoreArtifactRes(posnse). Otherwise it's not very clear.